skills_ml/algorithms/preprocessing/__init__.py

```diff
@@ -3,11 +3,55 @@
 import logging
 import inspect
 
+class ProcessingPipeline(object):
+    """A simple callable processing pipeline for imperative execution runtime.
+
+    This class will compose processing functions together to become a callable
+    object that takes in the input from the very first processing function and
+    returns the output of the last processing function.
+
+    Example:
+        This class can be used to create a callable vectorization object which
+        will transform a string into a vector and also preserve the preprocessing
+        functions for being reused later.
+        ```python
+        jp = JobPostingCollectionSample()
+        vectorization = ProcessingPipeline(
+            normalize,
+            clean_html,
+            clean_str,
+            word_tokenize,
+            partial(vectorize, embedding_model=w2v)
+        )
+
+        vector = vecotrization("Why so serious?")
+        ```
+
+     Attributes:
+        functions (generator): a series of functions
+
+    """
+    def __init__(self, *functions: Callable):
+        self.functions = functions
+
+    @property
+    def _compose(self):
```

Comment:
I have never seen the property decorator applied to a pseudo-private function before, so it seems weird. The property decorator is generally aimed at improving the external interface of an object. 

But to be honest, if you just want compose functionality, I would either move that to its own function that can be imported, or even just use  https://github.com/pytoolz/toolz

So this property goes away entirely and you have.


```
from toolz import compose
...
def __call__(self, input_to_be_processed):
    return compose(self.functions)(input_to_be_processed)
```

Reply:
finally have a reason to use pytoolz! yeah vote for that.